### PR TITLE
Upgrade ktlint gradle

### PR DESCRIPTION
**Pull request by andregasser**
# Description

Upgraded ktlint-gradle plugin to latest version 12.0.2. Along with this, code had to be reformatted in order to make the build work again. Major changed were:
- Bumped ktlint version to 12.0.2 in `gradle/libs.versions.toml` 
- Added `@Suppress("ktlint:standard:no-blank-line-in-list")` annotation to entity classes to make sure we can keep the formatting as it is.
- Added `@Suppress("ktlint:standard:annotation")` annotation to `Range` and `Scope` classes to ensure that the current formatting for the `@JvmOverloads` annotation can stay.
- Created a basic `.editorconfig` file which sets a couple of `ktlint` options.

All the remaining changes are formatting changes.

# Type of Change

- Dependency update / replacement

# Breaking Changes

None

# How Has This Been Tested?

Ran gradle build locally.

# Mandatory Checklist

- [x] I ran `gradle check` and there were no errors reported
- [x] I have performed a self-review of my code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] All tests pass locally with my changes
- [ ] I have added KDoc documentation to all public methods

# Optional Things To Check

The items below are some more things to check before asking other people to review your code.

- In case you worked on a new feature: Did you also implement the reactive endpoint (bigbone-rx)?
- In case you added new `*Methods` classes: Did you also reference it in the `MastodonClient` main class?
- Did you also update the documentation in the `/docs` folder (e.g. API Coverage page)?

### CODE REVIEW

File: bigbone-rx/src/main/kotlin/social/bigbone/rx/RxAnnouncementMethods.kt [hunk 1]
```diff
@@ -20,22 +20,15 @@ class RxAnnouncementMethods(client: MastodonClient) {
      * @see <a href="https://docs.joinmastodon.org/methods/announcements/#get">Mastodon API documentation: methods/announcements/#get</a>
      */
     @JvmOverloads
-    fun getAllAnnouncements(
-        withDismissed: Boolean = false
-    ): Single<List<Announcement>> = Single.fromCallable {
-        announcementMethods.getAllAnnouncements(withDismissed).execute()
-    }
+    fun getAllAnnouncements(withDismissed: Boolean = false): Single<List<Announcement>> =
+        Single.fromCallable { announcementMethods.getAllAnnouncements(withDismissed).execute() }
```
Comment (PattaFeuFeu) [inline bigbone-rx/src/main/kotlin/social/bigbone/rx/RxAnnouncementMethods.kt:24]: I really don’t prefer the new version 😕

File: bigbone-rx/src/main/kotlin/social/bigbone/rx/RxAppMethods.kt [hunk 2]
```diff
@@ -12,7 +12,6 @@ import social.bigbone.api.method.AppMethods
  * @see <a href="https://docs.joinmastodon.org/methods/apps/">Mastodon apps API methods</a>
  */
 class RxAppMethods(client: MastodonClient) {
-
     private val appMethods = AppMethods(client)
```
Comment (PattaFeuFeu) [inline bigbone-rx/src/main/kotlin/social/bigbone/rx/RxAppMethods.kt:15]: I also really don’t like the missing space. 😄

File: sample-kotlin/src/main/kotlin/social/bigbone/sample/ManageFilters.kt [hunk 3]
```diff
@@ -85,7 +88,10 @@ object ManageFilters {
      * @param client a [MastodonClient] with an authenticated user
      * @param filterId ID string for the filter that should be deleted
      */
-    private fun deleteFilter(client: MastodonClient, filterId: String) {
+    private fun deleteFilter(
+        client: MastodonClient,
+        filterId: String
```
Comment (PattaFeuFeu) [inline sample-kotlin/src/main/kotlin/social/bigbone/sample/ManageFilters.kt:93]: What’s the rule here? We use separate lines for methods with more than one parameter?

Comment (andregasser) [reply sample-kotlin/src/main/kotlin/social/bigbone/sample/ManageFilters.kt:93]: Yes, that's the ktlint standard rule for parameters.

File: bigbone-rx/src/main/kotlin/social/bigbone/rx/RxAppMethods.kt (hunk 2 above)
Comment (andregasser) [reply bigbone-rx/src/main/kotlin/social/bigbone/rx/RxAppMethods.kt:15]: Rule name here would be `standard:no-empty-first-line-in-class-body`.

Comment (PattaFeuFeu) [reply bigbone-rx/src/main/kotlin/social/bigbone/rx/RxAppMethods.kt:15]: How do you feel about that one? @andregasser @bocops 

I could live with the new version, but I know I’d write it differently if I were to write an application on my own.

File: bigbone-rx/src/main/kotlin/social/bigbone/rx/RxAnnouncementMethods.kt (hunk 1 above)
Comment (andregasser) [reply bigbone-rx/src/main/kotlin/social/bigbone/rx/RxAnnouncementMethods.kt:24]: Hey @PattaFeuFeu I also like the previous version better. @bocops any objections?

Comment (bocops) [reply bigbone-rx/src/main/kotlin/social/bigbone/rx/RxAnnouncementMethods.kt:24]: Hard to tell actually - I generally don't like having a single parameter on its own line if not absolutely necessary, but in this specific case, both declarations are somewhat crowded due to everything else that is going on at the same time.

Are any less complicated declarations affected by this rule change that we could have a look at? If not, no strong objection to reverting this, though.

File: bigbone-rx/src/main/kotlin/social/bigbone/rx/RxAppMethods.kt (hunk 2 above)
Comment (andregasser) [reply bigbone-rx/src/main/kotlin/social/bigbone/rx/RxAppMethods.kt:15]: Thanks for the feedback. I will disable the rule.

**Comment by PattaFeuFeu** [comment]
@andregasser It probably makes sense to block this PR until #408 #409 #414 #415 have been merged. Once those are merged, we are feature-complete and can then easily run the changes in this PR again.

Could you then please take care of rebasing this PR and applying the changes to everything on `main` once they’re merged?

**Comment by PattaFeuFeu** [comment]
@andregasser You wrote in #417 that this PR is now obsolete? I assume it can be closed then? 😊

**Comment by andregasser** [comment]
@PattaFeuFeu Correct, this one can be closed.

**Pull request closed without merging**